src/__tests__/basic.ts

```diff
@@ -17,6 +18,27 @@ function getPorts(count: number, options: PortFinderOptions = {}): Promise<numbe
     });
 }
 
+async function waitForPort(port: number): Promise<void> {
+    return new Promise<void>((resolve) => {
+        const tryConnect = () => {
+            const socket = new net.Socket();
+            socket.connect({ port, host: "localhost" }, () => {
+                socket.end();
+                resolve();
+            });
+
+            socket.on("error", (error) => {
+                // eslint-disable-next-line @typescript-eslint/no-explicit-any
+                if ((error as any).code === "ECONNREFUSED") {
+                    setTimeout(tryConnect, 50);
+                }
+            });
+        };
+
+        tryConnect();
+    });
+}
```

Comment:
maybe use https://www.npmjs.com/package/wait-on instead of implementing it on your own

Reply:
good idea